unity/Group55TestProjekt/Assets/Scripts/CellModel.cs

```diff
@@ -0,0 +1,253 @@
+    //     yp = Mathf.Min(Mathf.Max(yp + dy, 0), y);
+    // }
+
+    // private void CalculateB()
+    // {
+    //     float db;
+    //     db = (alpha2*kbar3*ap*(1-bp)-kbar5*bp)*k5;
+    //     bp = Mathf.Min(Mathf.Max(bp + db, 0), b);
+    // }
+
+    // private void MCP()
+    // { //Determine 'activity' of A based on ligands, cheR and cheB
+    //     //c and cheB decrease activity, cheR increases (methylation)
+    //     int nR = Binomial.Sample(pMR, cheR);
+    //     int nB = Binomial.Sample(pMB, cheBP);
+    //     float r = (float) nR / cheR;
+    //     float b = (float) nB / cheBP;
+    //     phi = (1-c)*(1-b)*r; 
+    //     int nA = Binomial.Sample(pA, (cheA-cheAP));
+    //     cheAP += nA;
+    // }
+
+    // private void CheA()
+    // { //Phosphorylation from A to B and Y
+
+    //     int nB = cheB-cheBP;
+    //     int nY = cheY-cheYP;
+    //     int n = Mathf.Min(cheAP, nB); //Available particles
+    //     int nP = Binomial.Sample(pAB,n); //transfers to B
+    //     cheBP += nP;
+    //     cheAP -= nP; //Subtract from A
+    //     n = Mathf.Min(cheAP, nY); //Available particles
+    //     nP = Binomial.Sample(pAY,n); //transfers to B
+    //     cheYP += nP; //portion to Y
+    //     cheAP -= nP; //Subtract from A
+    // }
+
+    // private void Flagella()
+    // {
+    //     int nP = Binomial.Sample(pFlagella,cheYP);
+    //     if(nP > 10)
+    //         this.run = false;
+    //     else
+    //         this.run = true;
+    // }
+
+    // private void CheY()
+    // { //Phosphorylated Y are auto dephosphorylated
+    //     if(cheYP > 0)
+    //     {
+    //         int nP = Binomial.Sample(pY,cheYP);
+    //         cheYP -= nP;
+    //     }
+    // }
+
+    // private void CheZ()
+    // { //Phosphorylated Y are dephosphorylated by Z
+    //     if(cheYP > 0)
+    //     {
+    //         int nP = Binomial.Sample(pZ,cheYP);
+    //         cheYP -= nP;
+    //     }
+    // }
+
+    // private void CheB()
+    // { //Phosphorylated B are auto dephosphorylated
+    //     if(cheBP > 0)
+    //     {
+    //         int nP = Binomial.Sample(pB,cheBP);
+    //         cheYP -= nP;
+    //     }
+    // }
+
+    private void CompareAndDecide()
+    { //Basic approach of just seeing if current position is better than last
+        int x = 5;
+        if( ( this.c / this.dc ) < 1 ) //Shift likelihood of tumbling
+            x += 2;
+        else
+            x -= 2;
+        float rand = Random.Range(0.0f,1.0f);
+        if( rand* 10 > x )
+            this.run = true;
+        else
+            this.run = false;
+    }
+
+    public bool IsRun()
+    {
+        this.dc = this.c;
+        this.c = model.environment.getConcentration(x, z);
+        DecideState();
+        bool state = this.run;
+        return state;
+    }
+
+    public void SetPos(float x, float z)
+    {
+        this.x = x;
+        this.z = z;
+    }
+
+    private void DecideState()
```

Comment:
Should this method not be removed since it just calls another private method

Reply:
Definitely needs more work, the currently used model is basically the old one and is just there so that we can run the architecture with something that displays OK behaviour. The other approaches that are commented out are still being balanced and refined to see their validity, but I think we should merge mostly for the changes in naming and communications between modules

Reply:
Ok sounds like a sound argument. Do you mind if I merge this into another branch where I have done some cleaning up and solve the merge conflicts there before adding it to dev?

Reply:
The point of keeping DecideState is so that we can restructure the logic of decision-making without changing the public-facing method

Reply:
And sure, that sounds good! I just think we would benefit from some cleaner structure, so that sounds like a good way to go